presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -103,9 +105,10 @@ public void process(Iterable<QueryExecution> queries)
         long totalBytes = 0;
         for (QueryExecution query : queries) {
             long bytes = query.getTotalMemoryReservation();
+            long queryMaxMemory = query.getMaxQueryMemory(maxQueryMemory.toBytes());
```

Comment:
this should be `Math.min(maxQueryMemory.toBytes(), query.getMaxQueryMemory())`. If you don't take the min with `maxQueryMemory` a user could set the limit above the cluster limit, which could cause a deadlock or heap space error